.github/workflows/ci.yml

```diff
@@ -1,6 +1,12 @@
 name: ci
 
-on: [push, pull_request]
+on:
+  pull_request:
+    branches:
+      - "*"
+  push:
+    branches:
+      - main
```

Comment:
@xyzshantaram what do you think about that?

Reply:
Doesn't branches: "*" do the same thing as it was doing before? I meant that it's better to run CI _only_ for PRs onto main, ie when we're getting ready for merge. Sorry I wasn't clearer.

Reply:
OK, I think the previous commit will do the job but I will rewrite it in a clearer way:

```
on:
  pull_request:
    branches:
      - main
  push:
    branches:
      - main
```